src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1092,7 +1092,7 @@ static bool matchFilter
 * Add a BSON filter based on a "q token".
 *
 */
-static void addBsonFilter
+static bool addBsonFilter
```

Comment:
This solution sufices for the case of q expressions in the GET /v2/entities. However, q expressions in subscriptions will be also problematic. 

This fact support the idea of do filter parsing at early stages, both for GET /v2/entities and subscriptions, and have the data as objects (i.e. https://github.com/telefonicaid/fiware-orion/issues/1705)

NTC (just a thought)